src/rmem.c

```diff
@@ -0,0 +1,470 @@
+#include "rmem.h"
+
+
+typedef uintptr_t ptrcmp;
+
+static size_t _AlignSize(const size_t size, const size_t align)
+{
+    return (size + (align-1)) & -align;
+}
+
+
+/************* Memory Pool *************/
+
+static void _RemoveNode(struct MemNode **const node)
+{
+    ((*node)->prev != NULL)? ((*node)->prev->next = (*node)->next) : (*node = (*node)->next);
```

Comment:
Some compilers might warn about the expression result being unused. Use an `if` statement instead.

Reply:
the only compilers that have good C99 support is the variants of GCC and Clang, both support this operation.

Reply:
Still, this is unnecessary crowded. Why not just use `if` statements like everyone else does?

Reply:
because this would become an `if`-`else` statement like the rest of the code :)